schema/crdb/dbinit.sql

```diff
@@ -1885,49 +1885,76 @@ CREATE INDEX IF NOT EXISTS lookup_console_by_silo_user ON omicron.public.console
 
 /*******************************************************************/
 
-CREATE TYPE IF NOT EXISTS omicron.public.update_artifact_kind AS ENUM (
-    -- Sled artifacts
-    'gimlet_sp',
-    'gimlet_rot',
-    'host',
-    'trampoline',
-    'control_plane',
+-- Describes a single uploaded TUF repo.
+--
+-- Identified by both a random uuid and its SHA256 hash. The hash could be the
+-- primary key, but it seems unnecessarily large and unwieldy.
+CREATE TABLE IF NOT EXISTS omicron.public.tuf_repo (
+    id UUID PRIMARY KEY,
+    time_created TIMESTAMPTZ NOT NULL,
+
+    sha256 STRING(64) NOT NULL,
+
+    -- The version of the targets.json role that was used to generate the repo.
+    targets_role_version INT NOT NULL,
+
+    -- The valid_until time for the repo.
+    valid_until TIMESTAMPTZ NOT NULL,
+
+    -- The system version described in the TUF repo.
+    --
+    -- Each system version is associated with just exactly one checksum.
+    system_version STRING(64) NOT NULL,
 
-    -- PSC artifacts
-    'psc_sp',
-    'psc_rot',
+    -- For debugging only:
+    -- Filename provided by the user.
+    file_name TEXT NOT NULL,
 
-    -- Switch artifacts
-    'switch_sp',
-    'switch_rot'
+    CONSTRAINT unique_checksum UNIQUE (sha256),
```

Comment:
Why must the checksum be unique? I don't think it's likely, but if we decided to publish a "new version, with no updated software", would the TUF repo have the same contents?

Or does this content also contain version info, so we'd always expect it to be unique, and this is just additional validation?

Reply:
My understanding is that the content would also contain version info, yeah.

Reply:
To confirm -- yes, `artifacts.json` contains the system version. When that changes, the hash will also change.